cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -961,6 +966,89 @@ public void deleteFlowQueues(HttpRequest request, HttpResponder responder,
     }
   }
 
+  @GET
+  @Path("/apps/{app-id}/services/{service-id}/routeconfig")
+  public void getRouteConfig(HttpRequest request, HttpResponder responder,
```

Comment:
It's better to add these methods to a new handler as they are not really program lifecycle.